### ISSUE TITLE
neomake and tagbar

### ISSUE THREAD

**Issue by nicodebo** [neomake/neomake#871]
### Issue

When using the [tagbar](https://github.com/majutsushi/tagbar) to jump to a function definition, neomake throw me the following error:

```
Error detected while processing function neomake#Make[9]..<SNR>113_Make:
line   80:
E741: Value is locked: jobinfo.maker.next = copy(a:options)
```

If I revert neomake back to commit https://github.com/neomake/neomake/commit/d9c02626a9e9fbcc04012966e00206e64c7ffb4e, the error is not triggered so commit https://github.com/neomake/neomake/commit/4c4767abb1178ed4f6d34466f94c0d1b6fe1b20e seems to be the first bad commit for me.

#### nvim version
NVIM 0.2.0-dev

#### neomake vimrc related lines
autocmd! BufWritePost,BufEnter * Neomake
let g:neomake_sh_enabled_makers = ['shellcheck']
let g:neomake_tex_enabled_makers = ['lacheck']

### Steps to reproduce

1. minimal working exemple
```
#!/usr/bin/env bash

function function1() {
    :
}

function function2() {
    :
}
```

2. Open the tagbar with ```:TagbarOpenAutoClose``` command
3. Select function1 or function2 tag and press enter.

#### Neomake debug information
Async support: 1
Current filetype: sh

##### Enabled makers
For the current filetype (with :Neomake):
 - shellcheck
   errorformat: '%f:%l:%c: %trror: %m,%f:%l:%c: %tarning: %m,%I%f:%l:%c: Note: %m'
   args: ['-fgcc']
   exe: 'shellcheck'
NOTE: you can define g:neomake_sh_enabled_makers to configure it (or b:neomake_sh_enabled_makers).

For the project (with :Neomake!): None.
NOTE: you can define g:neomake_enabled_makers to configure it.

Default maker settings:
 - remove_invalid_entries: 1
 - buffer_output: 1

##### Settings
```
g:neomake_tex_enabled_makers = ['lacheck']
g:neomake_place_signs = 1
g:neomake_sh_enabled_makers = ['shellcheck']


shell: zsh
shellcmdflag: -c
Windows: 0
```

#### :version
```

NVIM 0.2.0-dev
Build type: RelWithDebInfo
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -D_GNU_SOURCE -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/build/config -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/src -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/.deps/usr/include -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/.deps/usr/include -I/usr/include -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/build/src/nvim/auto -I/build/neovim-v3qzK5/neovim-0.1.5ubuntu1+git201612161508+2974+22~ubuntu16.04.1/build/include
Compiled by root@lcy01-22

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui      
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

```

#### :messages
```
"test.sh" 9L, 84C
Error detected while processing function neomake#Make[9]..<SNR>113_Make:
line   80:
E741: Value is locked: jobinfo.maker.next = copy(a:options)
```

### Additional information

* for python files, the error doesn't trigger
* for vim files, the error trigger
* for different bash of vim files, the number after ```<SNR>``` change. (I can provide if needed)

**Comment by yevhen-m** [comment]
I have a similar issue, but it jumps out when I try to commit via fugitive
```
Error detected while processing function neomake#Make[9]..<SNR>110_Make:
...
line   80:
E741: Value is locked: jobinfo.maker.next = copy(a:options)
```

**Comment by blueyed** [comment]
Please comment the lockvar for now.

**Comment by blueyed** [comment]
FWIW: this happens when an already running maker is detected (and to be restarted), probably through the `BufEnter` being invoked twice?!
Anyway, I have some WIP to fix this area (#867).
